examples/load_and_reindex_bboxes.py

```diff
@@ -0,0 +1,336 @@
+# between the two annotated values, and its width and height change linearly
+# as well.
+#
+# We use the dataset with NaN values as an input to the
+# ``interpolate_over_time`` function.
+ds_interp = ds_nan.copy()
+
+for data_array_str in ["position", "shape"]:
+    ds_interp[data_array_str] = interpolate_over_time(
+        data=ds_interp[data_array_str],
+        method="linear",
+        max_gap=None,
+        print_report=False,
+    )
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_interp.position.data[:14, 0, :])
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_interp.shape.data[:14, 0, :])
+
+
+# %%
+# Compare interpolation methods
+# ------------------------------
+# We can now qualitatively compare the three different methods of filling
+# in the missing frames, by plotting the bounding boxes
+# for the first 6 frames of the video.
+#
+# Remember only frames 0 and 5 are annotated in the original dataset. These
+# are plotted in blue, while the forward filled values are plotted in orange
+# and the linearly interpolated values in green.
+
+# sphinx_gallery_thumbnail_number = 2
+
+# initialise figure
+fig = plt.figure(figsize=(8, 8))
+
+# loop over frames
+for frame_n in range(6):
+    # add subplot axes
+    ax = plt.subplot(3, 2, frame_n + 1)
+
+    # plot frame
+    # note: the video is indexed at every frame, so
+    # we use the frame number as index
+    ax.imshow(video[frame_n])
+
+    # plot bounding box for each dataset
+    for ds_i, ds_one in enumerate([ds_nan, ds_ff, ds_interp]):
+        # plot box
+        top_left_corner = (
+            ds_one.position.sel(time=frame_n, individuals="id_1").data
+            - ds_one.shape.sel(time=frame_n, individuals="id_1").data / 2
+        )
+        bbox = plt.Rectangle(
+            xy=tuple(top_left_corner),
+            width=ds_one.shape.sel(
+                time=frame_n, individuals="id_1", space="x"
+            ).data,
+            height=ds_one.shape.sel(
+                time=frame_n, individuals="id_1", space="y"
+            ).data,
+            edgecolor=list_colors[ds_i],
+            facecolor="none",
+            # make line for NaN dataset thicker and dotted
+            linewidth=[5, 1.5, 1.5][ds_i],
+            linestyle=["dotted", "solid", "solid"][ds_i],
+            label=["nan", "ffill", "linear"][ds_i],
+        )
+        ax.add_patch(bbox)
+
+        # plot centroid
+        ax.scatter(
+            x=ds_one.position.sel(
+                time=frame_n, individuals="id_1", space="x"
+            ).data,
+            y=ds_one.position.sel(
+                time=frame_n, individuals="id_1", space="y"
+            ).data,
+            s=5,
+            color=list_colors[ds_i],
+        )
+
+    # add legend to first frame
+    if frame_n == 0:
+        ax.legend()
+    ax.set_title(f"Frame {frame_n}")
+    ax.set_xlabel("x (pixels)")
+    ax.set_ylabel("y (pixels)")
```

Comment:
If you adopt any of my suggestions I gave for the earlier plotting function, make sure this one follows the same conventions namings etc, for consistency.

Reply:
An aside on this plot, I find the yellow dot (centroid of ffil bbox) quite hard to see against the backdrop of the frame. Perhaps worth experimenting with a different color?

Reply:
I made the markers a bit bigger for visibility